app/assets/javascripts/discourse/app/lib/hashtag-autocomplete.js

```diff
@@ -10,59 +21,20 @@ import {
 } from "discourse/lib/utilities";
 import { INPUT_DELAY, isTesting } from "discourse-common/config/environment";
 import discourseDebounce from "discourse-common/lib/debounce";
-import domFromString from "discourse-common/lib/dom-from-string";
 import discourseLater from "discourse-common/lib/later";
 import { findRawTemplate } from "discourse-common/lib/raw-templates";
 
-let hashtagTypeClasses = {};
-export function registerHashtagType(type, typeClassInstance) {
-  hashtagTypeClasses[type] = typeClassInstance;
-}
-export function cleanUpHashtagTypeClasses() {
-  hashtagTypeClasses = {};
-}
-export function getHashtagTypeClasses() {
-  return hashtagTypeClasses;
-}
-export function decorateHashtags(element, site) {
-  element.querySelectorAll(".hashtag-cooked").forEach((hashtagEl) => {
-    // Replace the empty icon placeholder span with actual icon HTML.
-    const iconPlaceholderEl = hashtagEl.querySelector(
-      ".hashtag-icon-placeholder"
-    );
-    const hashtagType = hashtagEl.dataset.type;
-    const hashtagTypeClass = getHashtagTypeClasses()[hashtagType];
-    if (iconPlaceholderEl && hashtagTypeClass) {
-      const hashtagIconHTML = hashtagTypeClass
-        .generateIconHTML({
-          icon: site.hashtag_icons[hashtagType],
-          id: hashtagEl.dataset.id,
-        })
-        .trim();
-      iconPlaceholderEl.replaceWith(domFromString(hashtagIconHTML)[0]);
-    }
-
-    // Add an aria-label to the hashtag element so that screen readers
-    // can read the hashtag text.
-    hashtagEl.setAttribute("aria-label", `${hashtagEl.innerText.trim()}`);
-  });
-}
-
-export function generatePlaceholderHashtagHTML(type, spanEl, data) {
-  // NOTE: When changing the HTML structure here, you must also change
-  // it in the hashtag-autocomplete markdown rule, and vice-versa.
-  const link = document.createElement("a");
-  link.classList.add("hashtag-cooked");
-  link.href = data.relative_url;
-  link.dataset.type = type;
-  link.dataset.id = data.id;
-  link.dataset.slug = data.slug;
-  const hashtagTypeClass = new getHashtagTypeClasses()[type];
-  link.innerHTML = `${hashtagTypeClass.generateIconHTML(
-    data
-  )}<span>${emojiUnescape(data.text)}</span>`;
-  spanEl.replaceWith(link);
-}
+// TODO (martin) Remove this once plugins have changed to use hashtag-decorator and
```

Comment:
Wrappers that call `deprecated` would be better to inform current users of this.

Reply:
Sure, I am pretty sure discourse-encrypt is the only thing that uses these, but will wrap in decorators anyway :+1: